src/verifier.cc

```diff
@@ -177,18 +188,204 @@ void SetupToleranceForTexels(const ProbeCommand* command,
   }
 }
 
+// Convert data of |texel| into double values based on the
+// information given in |texel_format|.
+std::vector<Value> GetActualValuesFromTexel(const uint8_t* texel,
+                                            const Format* texel_format) {
+  assert(texel_format && !texel_format->GetComponents().empty());
+
+  std::vector<Value> actual_values(texel_format->GetComponents().size());
+  uint8_t bit_offset = 0;
+
+  for (size_t i = 0; i < texel_format->GetComponents().size(); ++i) {
+    const auto& component = texel_format->GetComponents()[i];
+    uint8_t actual[8] = {};
+
+    BitCopy::CopyMemoryToBuffer(actual, texel, bit_offset, component.num_bits);
+    if (component.mode == FormatMode::kUFloat ||
+        component.mode == FormatMode::kSFloat) {
+      if (component.num_bits < 32) {
+        actual_values[i].SetDoubleValue(static_cast<double>(
+            BitCopy::HexFloatToFloat(actual, component.num_bits)));
+      } else if (component.num_bits == 32) {
+        float* ptr = reinterpret_cast<float*>(actual);
+        actual_values[i].SetDoubleValue(static_cast<double>(*ptr));
+      } else if (component.num_bits == 64) {
+        double* ptr = reinterpret_cast<double*>(actual);
+        actual_values[i].SetDoubleValue(*ptr);
+      } else {
+        assert(false && "Bits of component is not for double nor float type");
+      }
+    } else {
+      uint16_t* ptr16 = nullptr;
+      uint32_t* ptr32 = nullptr;
+      uint64_t* ptr64 = nullptr;
+      switch (component.num_bits) {
+        case 8:
+          actual_values[i].SetDoubleValue(static_cast<double>(*actual));
```

Comment:
Why are these all set as doubles and not integers?

Reply:
We had offline discussion and determined to keep it.